test/e2e/client.go

```diff
@@ -127,6 +139,42 @@ func (c *E2ETestClient) Get(ctx context.Context, key types.NamespacedName, obj c
 	return c.Client.Get(ctx, key, obj)
 }
 
+func (c *E2ETestClient) Execute(pod corev1.Pod, containerName, command string) (string, error) {
+	req := c.CoreV1Client.RESTClient().
+		Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: containerName,
+			Command:   []string{"/bin/sh", "-c", command},
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       true,
+		}, scheme.ParameterCodec)
+
+	buf := &bytes.Buffer{}
+	errBuf := &bytes.Buffer{}
+	exec, err := remotecommand.NewSPDYExecutor(c.restConfig, "POST", req.URL())
+	if err != nil {
+		panic(err)
```

Comment:
[q] why panic here when error is already part of method signature, is there a specific reason to panic instead of return the error?

Reply:
good catch! Just bad copy-paste 🤦